### ISSUE TITLE
Tensorflow 2.2.0 does not save metrics using model.compile(metrics=['accuracy'])

### ISSUE THREAD

**Issue by MalteEbner** [tensorflow/tensorflow#37714]
Minimal example for bug:

```
import tensorflow as tf
from tensorflow.keras import layers

model = tf.keras.Sequential([
layers.Dense(64, activation='relu', input_shape=(32,)),
layers.Dense(10)])

model.compile(optimizer=tf.keras.optimizers.Adam(0.01),
              loss=tf.keras.losses.CategoricalCrossentropy(from_logits=True),
              metrics=['accuracy'])

print(model.metrics)
print(model.metrics_names)
```

output with tensorflow 2.1.0:
[<tensorflow.python.keras.metrics.MeanMetricWrapper object at 0x1497f1400>]
['loss', 'accuracy']

output with tensorflow 2.2.0rc0:
[]
[]

**Comment by jaketae** [comment]
Thank you for the issue. I was able to reproduce this error. Take a look at the [gist](https://gist.github.com/jaketae/b02028462cf17b8782bb826ae30c6f5d.js).

**Comment by howl-anderson** [comment]
@MalteEbner TF 2.2 changes the code, you need call `fit` or `train_on_batch` first then you can get `model.metrics` or `model.metrics_names` correctly, see [this colab](https://colab.research.google.com/drive/1AunEdHAAUZPbH0q4aGsZQ44hL1WKoiOr)

**Comment by ravikyram** [comment]
@MalteEbner 

Is this still an issue?. Please close this thread if it solves your question. Thanks!

**Comment by thimabru1010** [comment]
> Minimal example for bug:
> 
> ```
> import tensorflow as tf
> from tensorflow.keras import layers
> 
> model = tf.keras.Sequential([
> layers.Dense(64, activation='relu', input_shape=(32,)),
> layers.Dense(10)])
> 
> model.compile(optimizer=tf.keras.optimizers.Adam(0.01),
>               loss=tf.keras.losses.CategoricalCrossentropy(from_logits=True),
>               metrics=['accuracy'])
> 
> print(model.metrics)
> print(model.metrics_names)
> ```
> 
> output with tensorflow 2.1.0:
> [<tensorflow.python.keras.metrics.MeanMetricWrapper object at 0x1497f1400>]
> ['loss', 'accuracy']
> 
> output with tensorflow 2.2.0rc0:
> []
> []

got this too
